src/tokenizer.ts

```diff
@@ -0,0 +1,45 @@
+const REPEATABLE_SYMBOLS = [
+  "+",
+  "-",
+  "*",
+  "/",
+  "=",
+  "<",
+  ">",
+  "_",
+  "#",
+  ".",
+  "|",
+  "&",
+];
+
+const FIXED_TOKENS = ["!==", "!=", "+=", "-=", "*=", "/=", "%=", "<=", ">="];
+const REPEATABLE_SYMBOLS_REGEX = REPEATABLE_SYMBOLS.map(escapeRegExp)
+  .map((s) => `${s}+`)
+  .join("|");
+const FIXED_TOKENS_REGEX = FIXED_TOKENS.map(escapeRegExp).join("|");
+const IDENTIFIERS_REGEX = "[a-zA-Z_]+[a-zA-Z_0-9]*";
+const SINGLE_SYMBOLS_REGEX = "[^\\s\\w]";
+const NUMBERS_REGEX = "(?<![.])\\d+.\\d+(?![.])|\\d+"; // (not-dot digits dot digits not-dot) OR digits
```

Comment:
Any performance issues with the look ahead? I know these can be a bit slower but I'd imagine not enough to be noticeable here

Reply:
I haven't noticed anything but I haven't measured either.